hw8/todolist/src/components/Task/TaskComponent.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import "./TaskComponent.css";
+import { Task } from "../../interfaces/Task";
+import {
+  setCurrentTaskId,
+  setModalType,
+  toggleModalVisibility,
+} from "../../store/taskSlice";
+import { useAppDispatch } from "../../store/hooks";
+
+interface Props {
+  task: Task;
+  onDelete: (currentTask: Task) => void;
+  onComplete?: (currentTask: Task) => void;
+}
+
+const displayDate = (date: Date) => {
+  const correctDate = new Date(date);
+  const today = new Date();
+  const tomorrow = new Date(
+    today.getFullYear(),
+    today.getMonth(),
+    today.getDate() + 1
+  );
+  if (correctDate.toDateString() === today.toDateString()) {
+    return "Today";
+  }
+  if (correctDate.toDateString() === tomorrow.toDateString()) {
+    return "Tomorrow";
+  }
+  return correctDate.toDateString();
+};
+
+const TaskComponent = ({ task, onDelete, onComplete }: Props) => {
+  const { title, tag, deadline } = task;
+  const type = task.isCompleted ? "completed" : "current";
+
+  const dispatch = useAppDispatch();
+
+  return (
+    <div className={`row ${type}`}>
+      <button
+        className={`${type}__checkbox`}
+        onClick={() => {
+          if (onComplete) onComplete(task);
+        }}
+      ></button>
+      <div className={`column ${type}__main-text`}>
```

Comment:
It's also better to extract it to the separated component. There's too much <div> tags and it's hard to understand what's the element here